hnix-store-remote/src/System/Nix/Store/Remote.hs

```diff
@@ -32,3 +35,25 @@ verifyStore :: CheckFlag -> RepairFlag -> MonadStore Bool
 verifyStore check repair = simpleOpArgs VerifyStore $ do
   putBool check
   putBool repair
+
+computeClosure
+  :: (Monad m, Foldable f, Monoid a)
+  => (a -> m (f r))
+  -> (r -> a)
+  -> a
+  -> m a
+computeClosure lookupReferences findReferences = go
```

Comment:
It's not awful but I feel like there might be something I'm missing to simplify this.